recipes/glslang/all/conanfile.py

```diff
@@ -127,29 +114,15 @@ def generate(self):
         deps.generate()
 
     def _patch_sources(self):
-        apply_conandata_patches(self)
-        # Do not force PIC if static (but keep it if shared, because OGLCompiler, OSDependent,
-        # GenericCodeGen and MachineIndependent are still static and linked to glslang shared)
-        if not self.options.shared:
-            cmake_files_to_fix = [
-                {"target": "OGLCompiler", "relpath": os.path.join("OGLCompilersDLL", "CMakeLists.txt")},
-                {"target": "SPIRV"      , "relpath": os.path.join("SPIRV", "CMakeLists.txt")},
-                {"target": "SPVRemapper", "relpath": os.path.join("SPIRV", "CMakeLists.txt")},
-                {"target": "OSDependent", "relpath": os.path.join("glslang", "OSDependent", "Unix","CMakeLists.txt")},
-                {"target": "OSDependent", "relpath": os.path.join("glslang", "OSDependent", "Windows","CMakeLists.txt")},
-                {"target": "HLSL"       , "relpath": os.path.join("hlsl", "CMakeLists.txt")},
-            ]
-            glslang_version = Version(self.version)
-            if glslang_version >= "7.0.0" and glslang_version < "11.0.0":
-                cmake_files_to_fix.append({"target": "glslang", "relpath": os.path.join("glslang", "CMakeLists.txt")})
-            else:
-                cmake_files_to_fix.append({"target": "glslang-default-resource-limits", "relpath": os.path.join("StandAlone" , "CMakeLists.txt")})
-                cmake_files_to_fix.append({"target": "MachineIndependent", "relpath": os.path.join("glslang", "CMakeLists.txt")})
-                cmake_files_to_fix.append({"target": "GenericCodeGen", "relpath": os.path.join("glslang", "CMakeLists.txt")})
-            for cmake_file in cmake_files_to_fix:
-                replace_in_file(self, os.path.join(self.source_folder, cmake_file["relpath"]),
-                                      "set_property(TARGET {} PROPERTY POSITION_INDEPENDENT_CODE ON)".format(cmake_file["target"]),
-                                      "")
+        for cmake_file in sorted(self.source_path.rglob("CMakeLists.txt")):
+            content = cmake_file.read_text(encoding="utf8")
+            if "POSITION_INDEPENDENT_CODE ON" in content:
+                content = re.sub(r"set_property\(TARGET \S+ PROPERTY POSITION_INDEPENDENT_CODE ON\)\n", "", content)
+                content = content.replace("POSITION_INDEPENDENT_CODE ON", "")
+                cmake_file.write_text(content, encoding="utf8")
+                self.output.info(f"Patched fPIC handling in {cmake_file.relative_to(self.source_path)}")
+            if "POSITION_INDEPENDENT_CODE" in content:
+                raise ConanException(f"POSITION_INDEPENDENT_CODE found in {cmake_file}, please update the recipe")
```

Comment:
This needs to be wrapped in `if not self.options.shared:`

I tested 1.3.239.0 in all 3 configs (shared, fPIC=True/False), all worked. The only thing I'm not sure about is that the previous version unconditionally patched the top level CMakeLists.txt, this would now do it conditionally.